src/WebJobs.Script/Workers/Rpc/Configuration/WorkerConfigurationResolver.cs

```diff
@@ -0,0 +1,269 @@
+                        {
+                            continue;
+                        }
+                    }
+
+                    string workerConfigPath = Path.Combine(workerPath, RpcWorkerConstants.WorkerConfigFileName);
+                    if (File.Exists(workerConfigPath))
+                    {
+                        outputDict[workerDir] = workerPath;
+                    }
+
+                    if (!_environment.IsMultiLanguageRuntimeEnvironment() &&
+                        workerRuntime is not null &&
+                        outputDict.ContainsKey(workerRuntime))
+                    {
+                        break;
+                    }
+                }
+            }
+        }
+
+        private static IEnumerable<Version> ParseWorkerVersionsInDescending(IEnumerable<string> workerVersions)
+        {
+            var versions = new List<Version>();
+
+            foreach (var workerVersion in workerVersions)
+            {
+                string versionDir = Path.GetFileName(workerVersion);
+                string formattedVersion = FormatVersion(versionDir);
+
+                if (Version.TryParse(formattedVersion, out Version version))
+                {
+                    versions.Add(version);
+                }
+            }
+
+            return versions.OrderDescending();
+        }
+
+        private static string FormatVersion(string version)
+        {
+            if (!version.Contains('.'))
+            {
+                version = version + ".0"; // Handle versions like '1' as '1.0'
+            }
+
+            return version;
+        }
+
+        private bool IsWorkerCompatibleWithHost(string workerDir)
+        {
+            string workerConfigPath = Path.Combine(workerDir, RpcWorkerConstants.WorkerConfigFileName);
+            if (!File.Exists(workerConfigPath))
+            {
+                return false;
+            }
+
+            JsonElement workerConfig = WorkerConfigurationHelper.GetWorkerConfigJsonElement(workerConfigPath);
+
+            // static capability resolution
+            bool doesHostRequirementMeet = DoesHostRequirementMeet(workerConfig);
+
+            if (!doesHostRequirementMeet)
+            {
+                return false;
+            }
+
+            // profiles evaluation
+            RpcWorkerDescription workerDescription = WorkerConfigurationHelper.GetWorkerDescription(
+                workerConfig,
+                _jsonSerializerOptions,
+                workerDir,
+                _profileManager,
+                _config,
+                _logger,
+                true,
+                _workersAvailableForResolutionViaHostingConfig);
+
+            if (workerDescription.IsDisabled == true)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private HashSet<string> GetHostRequirements(JsonElement workerConfig)
```

Comment:
Can you add a method summary with an example of `hostRequirements`? That helps to visualize the flow when reviewing this.